test/Tests/src/Network/Http_Spec.enso

```diff
@@ -1,332 +1,205 @@
-            req = Request.new HTTP_Method.Post url_post
-            req_with_body = req.with_json json_text
-            res = HTTP.new.request req_with_body
-            res.code.should_equal HTTP_Status_Code.ok
-            res.body.decode_as_json.should_equal expected_response
+            response = HTTP.post url_post test_file
+            response.at "data" . should_equal test_file.read_text
+
+        Test.specify "Can perform a Text POST with auto-conversion" <|
+            response = HTTP.post url_post "hello world"
+            response.at "data" . should_equal "hello world"
+            response.at "headers" . at "Content-Length" . should_equal "11"
+            response.at "headers" . at "Content-Type" . should_equal "text/plain"
+
+        Test.specify "Can skip auto-parse" <|
+            response = HTTP.post url_post (Request_Body.Text "hello world") try_auto_parse_response=False
+            response.body.bytes.length . should_equal 383
+
+        Test.specify "Can send a custom header" <|
+            response = HTTP.post url_post (Request_Body.Text "hello world") headers=[Header.new "Custom" "asdf"]
+            response.at "data" . should_equal "hello world"
+            response.at "headers" . at "Content-Length" . should_equal "11"
+            response.at "headers" . at "Content-Type" . should_equal "text/plain"
+            response.at "headers" . at "Custom" . should_equal "asdf"
+
+        Test.specify "Cannot POST when output context is disabled" <|
+            Context.Output.with_disabled <|
+                HTTP.post url_post (Request_Body.Text "hello world") . should_fail_with Forbidden_Operation
+
+        Test.specify "Can perform a Request_Body.Text PUT" <|
+            url_put = base_url_with_slash + "put"
+            response = HTTP.post url_put (Request_Body.Text "hello world") method=HTTP_Method.Put
+            response.at "data" . should_equal "hello world"
+            response.at "headers" . at "Content-Length" . should_equal "11"
+            response.at "headers" . at "Content-Type" . should_equal "text/plain"
+
+        Test.specify "Can perform a Request_Body.Text PATCH" <|
+            url_patch = base_url_with_slash + "patch"
+            response = HTTP.post url_patch (Request_Body.Text "hello world") method=HTTP_Method.Patch headers=[Header.content_type "application/diff"]
+            response.at "data" . should_equal "hello world"
+            response.at "headers" . at "Content-Length" . should_equal "11"
+            response.at "headers" . at "Content-Type" . should_equal "application/diff"
+
+        Test.specify "Unsupported method" <|
+            url_get = base_url_with_slash + "get"
+            HTTP.post url_get (Request_Body.Text "hello world") method=HTTP_Method.Get . should_fail_with Illegal_Argument
+
+        Test.specify "unknown host" <|
+            HTTP.post "http://undefined_host.invalid" (Request_Body.Text "hello world") . should_fail_with Request_Error
```

Comment:
We are missing tests for what happens if the output context is disabled.

I assume POST etc. should fail with `Forbidden_Operation` - we should have tests for that.

I guess GET is supposed to work even with context disabled? In that case, we should have a test as well - I'd suggest to use a pattern `run_with_and_without_output` like the one I used in Database tests (you can move the file).

Reply:
We also absolutely need a test ensuring that `DELETE` method **does not run** with output context disabled, as that is also a side-effecting and likely destructive operation - so it should only run in enabled context.

Reply:
There is a test for disabled output context here: https://github.com/enso-org/enso/pull/7700/files/517e0cafc15e66265532882be686ebd8fcb5fff0#diff-6f89099e49e3c36269e473f0f516f35d39c196340b7f8d49392edfae615dadd7R176

Since I'm preventing *all* methods in this case, at the very top, I have just one test.

Reply:
My comment about DELETE was valid as long as it was a `fetch` method and not on `post`.
Still, just to make sure we don't accidentally change something - I'd add a test that specifically rules out the DELETE method on `fetch`. I'd just add one more test case to https://github.com/enso-org/enso/pull/7700/files/517e0cafc15e66265532882be686ebd8fcb5fff0#diff-6f89099e49e3c36269e473f0f516f35d39c196340b7f8d49392edfae615dadd7R196